### PR TITLE
feat: Adding `examples/single-sign-on`.

### LINKED ISSUE #433: [BLUEPRINT] - External IDP for SSO
Add a blueprint which demonstrates how to leverage and external identity provider (IDP) to govern cluster access with EKS Blueprints.

### PR DESCRIPTION
**Pull request by rodrigobersa**
# Description

This example allows users to integrate IAM Identity Center and Okta as external IdP providers for Single Sign-On solution.

### Motivation and Context

- Resolves #433 

### How was this change tested?

- [X] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [X] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [X] Yes, I ran `pre-commit run -a` with this PR

### CODE REVIEW

File: examples/single-sign-on/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Single Sign-On for Amazon EKS Cluster
+
+These examples demonstrates how to deploy an Amazon EKS cluster that is deployed on the AWS Cloud, integrated with an external Identity Provider (IdP) for Single Sign-On (SSO) authentication. The authorization configuration still being done using Kubernetes Role-based access control (RBAC). By the this time we have integration with the following IdPs.
```
Comment (askulkarni2) [inline examples/single-sign-on/README.md:3]:
```suggestion
These examples demonstrates how to deploy an Amazon EKS cluster that is deployed on the AWS Cloud, integrated with an external Identity Provider (IdP) for Single Sign-On (SSO) authentication. The authorization configuration still being done using Kubernetes Role-based access control (RBAC). At this time we have integration with the following IdPs.
```

File: examples/single-sign-on/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+To provision these examples, run the following commands:
+
+```sh
+terraform init
+<<<<<<< HEAD
+terraform apply -target module.vpc
+=======
+terraform apply -target module.vpc 
+>>>>>>> 411e9828 (Adding  example)
```
Comment (askulkarni2) [inline examples/single-sign-on/README.md:26]:
```suggestion
terraform init
terraform apply -target module.vpc
```

File: examples/single-sign-on/iam-identity-center/README.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
+# IAM Identity Center Single Sign-On for Amazon EKS Cluster
+
+These example demonstrates how to deploy an Amazon EKS cluster that is deployed on the AWS Cloud, integrated with IAM Identity Center (former AWS SSO) as an the Identity Provider (IdP) for Single Sign-On (SSO) authentication. The authorization configuration layer still being done using Kubernetes Role-based access control (RBAC). By the this time we have integration with the following IdPs.
```
Comment (askulkarni2) [inline examples/single-sign-on/iam-identity-center/README.md:3]:
```suggestion
This example demonstrates how to deploy an Amazon EKS cluster that is deployed on the AWS Cloud, integrated with IAM Identity Center (former AWS SSO) as an the Identity Provider (IdP) for Single Sign-On (SSO) authentication. The configuration for authorization is done using Kubernetes Role-based access control (RBAC).
```

File: examples/single-sign-on/iam-identity-center/README.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+terraform apply
+```
+
+Enter `yes` at command prompt to apply
+
+## Validate
+
+After the `terraform` commands are executed sucessfully, check if the newly created users are active.
+
+To do that use the link provided in the email invite if you added a valid email address for your users, or go to the [IAM Identity Center Console](https://console.aws.amazon.com/singlesignon/home/), in the *Users* dashboard on the left hand side menu, then select the user, and click on *Reset password* button on the upper right corner. Choose the option to *Generate a one-time password and share the password with the user*.
```
Comment (askulkarni2) [inline examples/single-sign-on/iam-identity-center/README.md:35]: Where should I add email addresses for my users? I am not sure its clear if this is something I do via Terraform or AWS Console.

File: examples/single-sign-on/iam-identity-center/sso.tf [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+resource "aws_identitystore_user" "user" {
+  identity_store_id = tolist(data.aws_ssoadmin_instances.this.identity_store_ids)[0]
+  display_name      = "Developer User"
+  user_name         = "user@example.com"
+
+  name {
+    family_name = "User"
+    given_name  = "Developer"
+  }
+}
```
Comment (askulkarni2) [inline examples/single-sign-on/iam-identity-center/sso.tf:125]: Would it make sense to accept a map of User configuration as a variable?

File: examples/single-sign-on/iam-identity-center/sso.tf [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+    api_group = "rbac.authorization.k8s.io"
+    kind      = "ClusterRole"
+    name      = "view"
+  }
+  subject {
+    kind      = "Group"
+    name      = "eks-developers"
+    api_group = "rbac.authorization.k8s.io"
+  }
+}
```
Comment (askulkarni2) [inline examples/single-sign-on/iam-identity-center/sso.tf:203]: I am wondering if it makes sense to show how to use teams module here and tie in SSO groups to teams/namespaces?

File: examples/single-sign-on/okta/README.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
+# Okta Single Sign-On for Amazon EKS Cluster
+
+These example demonstrates how to deploy an Amazon EKS cluster that is deployed on the AWS Cloud, integrated with Okta as an the Identity Provider (IdP) for Single Sign-On (SSO) authentication. The authorization configuration layer still being done using Kubernetes Role-based access control (RBAC). By the this time we have integration with the following IdPs.
```
Comment (askulkarni2) [inline examples/single-sign-on/okta/README.md:3]:
```suggestion
This example demonstrates how to deploy an Amazon EKS cluster that is deployed on the AWS Cloud, integrated with Okta as an the Identity Provider (IdP) for Single Sign-On (SSO) authentication. The configuration for authorization is done using Kubernetes Role-based access control (RBAC).
```

File: examples/single-sign-on/okta/okta.tf [hunk 8]
```diff
@@ -0,0 +1,155 @@
+# You can generate an Okta API token in the Okta Developer Console. Follow these instructions: https://bit.ly/get-okta-api-token
```
Comment (askulkarni2) [inline examples/single-sign-on/okta/okta.tf:1]: Same note regarding usage of teams module and showing how groups can be tied to namespaces?

File: examples/single-sign-on/okta/outputs.tf [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+output "okta_login" {
+  description = "Setup OIDC Login for OKTA."
+  value       = "kubectl oidc-login setup --oidc-issuer-url=${okta_auth_server.eks.issuer} --oidc-client-id=${okta_app_oauth.eks.id}"
+}
+
+output "configure_kubeconfig" {
+  description = "Update kubeconfig with OKTA OIDC parameters."
+  value       = <<EOT
+    kubectl config set-credentials oidc \
```
Comment (askulkarni2) [inline examples/single-sign-on/okta/outputs.tf:14]: Nice!

**Review by askulkarni2 — COMMENTED** [review]
@rodrigobersa looks great. Some comments before proceeding.

File: examples/single-sign-on/README.md (hunk 1 above)
Comment (rodrigobersa) [reply examples/single-sign-on/README.md]: Fixed!

File: examples/single-sign-on/iam-identity-center/README.md (hunk 3 above)
Comment (rodrigobersa) [reply examples/single-sign-on/iam-identity-center/README.md]: Fixed!

File: examples/single-sign-on/okta/README.md (hunk 7 above)
Comment (rodrigobersa) [reply examples/single-sign-on/okta/README.md]: Fixed!

File: examples/single-sign-on/okta/outputs.tf (hunk 9 above)
Comment (rodrigobersa) [reply examples/single-sign-on/okta/outputs.tf:14]: Thanks!

File: examples/single-sign-on/iam-identity-center/sso.tf (hunk 5 above)
Comment (rodrigobersa) [reply examples/single-sign-on/iam-identity-center/sso.tf:125]: I think yes, my concern is to the example become "consumable".

File: examples/single-sign-on/iam-identity-center/sso.tf (hunk 6 above)
Comment (rodrigobersa) [reply examples/single-sign-on/iam-identity-center/sso.tf:203]: Of course! I can do it.

File: examples/single-sign-on/README.md (hunk 2 above)
Comment (rodrigobersa) [reply examples/single-sign-on/README.md]: Didn't see that coming. Fixed.

File: examples/single-sign-on/iam-identity-center/sso.tf (hunk 5 above)
Comment (rodrigobersa) [reply examples/single-sign-on/iam-identity-center/sso.tf]: Done! :)

File: examples/single-sign-on/iam-identity-center/sso.tf (hunk 6 above)
Comment (rodrigobersa) [reply examples/single-sign-on/iam-identity-center/sso.tf]: Done! :)

File: examples/single-sign-on/iam-identity-center/README.md (hunk 4 above)
Comment (rodrigobersa) [reply examples/single-sign-on/iam-identity-center/README.md]: Fixed.

File: examples/single-sign-on/okta/okta.tf (hunk 8 above)
Comment (rodrigobersa) [reply examples/single-sign-on/okta/okta.tf:1]: Done! :)

**Review by askulkarni2 — APPROVED** [review]
LGTM!

**Pull request merged**